packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -683,28 +683,29 @@ abstract class AssetBundleImageProvider extends ImageProvider<AssetBundleImageKe
   }
 }
 
+/// Key for the image obtained by a [ResizeImage].
+///
+/// This is used to identify the precise resource in the [imageCache].
 @immutable
-class _SizeAwareCacheKey {
-  const _SizeAwareCacheKey(this.providerCacheKey, this.width, this.height);
-
-  final Object providerCacheKey;
+class ResizeImageKey {
```

Comment:
This being public is actively dangerous, I believe -- it means that this code can't assume that the cache wasn't poisoned by other code masquerading using these private types.

Reply:
Outside of this file nobody can instantiate a ResizeImageKey because the constructor is private. So, i don't think you can poison the cache. I'll add a comment to the code about this.

Reply:
Well, basically, what you said in your next comment 😄

Reply:
but you could still implement them